src/Controller/TagsController.php

```diff
@@ -246,4 +246,33 @@ public function search()
             $search
         ]);
     }
+
+    /**
+     * Display list of tags for autocompletion.
+     *
+     * @param String $filter Filters the tags list with only those that contain the
+     *                       search string.
+     */
+    public function autocomplete($search)
+    {
+        $this->helpers[] = 'Tags';
+
+        $conditions = [];
+        if (!empty($search)) {
+            $conditions = [
+                'name LIKE' => "$search%"
+            ];
+        }
+        $this->paginate = [
+            'limit' => 10,
+            'fields' => ['name', 'id', 'nbrOfSentences'],
+            'order' => ['nbrOfSentences' => 'DESC'],
+            'conditions' => $conditions
+        ];
+
+        $this->autoRender = false;
+        $this->response = $this->response->withType('json');
+        $allTags = $this->paginate();
+        echo json_encode($allTags);
```

Comment:
If you’re trying to echo something, this should happen in the View, not the Controller. :point_up_2: Otherwise, errors like "headers already sent" are likely to happen.

In addition, if all you need to do is to echo the results in json, you can [do it from the controller](https://book.cakephp.org/3/fr/views/json-and-xml-views.html), like this:
```
$this->loadComponent('RequestHandler');
$this->set('allTags', $allTags);
$this->set('_serialize', ['allTags']);
$this->RequestHandler->renderAs($this, 'json');
```

Reply:
Thank you, this makes more sense, I'm new to CakePHP so I have to practise :+1: